feat: support file system caching of trace files

DavePearce opened the pull request:
## Import Notes

* (**Testing**) As yet, the plugin `GenerateConflatedTracesV2` has not been tested.  However, the `LtTraceFile` class has.

* (**Acivation**) There is no command-line argument (or similar) for activating or deactiving caching.  This could be done using a CLI argument for the `TracesEndpointServicePlugin`.

DavePearce committed:
preliminary implementation of tracefile caching

This puts in place a preliminary (and completely untested)
implementation of tracefile caching.  In particular, this attempts to
read the existing tracefile (if it exists) and check the embedded
version number.  If all of these checks pass, it will reuse the trace
file.

DavePearce committed:
rework tracefile parser

This reworks the tracefile parsing logic to be more robust.  For
example, if we want to extract other metadata attributes.  Also, it has
now been tested on some real trace files.

DavePearce commented on arithmetization/src/main/java/net/consensys/linea/plugins/rpc/tracegeneration/GenerateConflatedTracesV2.java:
```diff
@@ -100,33 +104,85 @@ private TraceFile generateTraceFile(PluginRpcRequest request) {
+              params.startBlockNumber(),
+              params.endBlockNumber(),
+              params.expectedTracesEngineVersion());
+      log.info("[TRACING] trace for {}-{} serialized to {} in {}", path, fromBlock, toBlock, sw);
```
Fixed

DavePearce commented on arithmetization/src/main/java/net/consensys/linea/zktracer/LtTraceFile.java:
```diff
@@ -0,0 +1,213 @@
+      //
+      return metadataBytes;
+    }
+  }
```
fixed

FlorianHuc committed:
Add option to CLI

FlorianHuc committed:
remove required as default is set.

FlorianHuc commented on arithmetization/src/main/java/net/consensys/linea/plugins/rpc/tracegeneration/GenerateConflatedTracesV2.java:
```diff
@@ -100,33 +105,87 @@ private TraceFile generateTraceFile(PluginRpcRequest request) {
+    }
+    // Read trace file header
+    try (LtTraceFile tf = new LtTraceFile(path)) {
+      LtTraceFile.Header header = tf.getHeader();
```
given the file name contains both besu and plugin version, we could actually skip the header validation no?
What it provides is that it's well formed.

DavePearce committed:
remove LtTraceFile header check

The header check is unnecessary as the releaseVersion is already
embedded in the filename.  Therefore, if the file exists, we know its
from the right releaseVersion.
